app/src/main/kotlin/at/bitfire/davdroid/resource/LocalDataStore.kt

```diff
@@ -51,4 +51,12 @@ interface LocalDataStore<T: LocalCollection<*>> {
      */
     fun delete(localCollection: T)
 
+    /**
+     * Changes the account assigned to the containing data to another one.
+     *
+     * @param oldAccount The old account.
+     * @param newAccount The new account.
+     */
+    fun updateAccount(provider: ContentProviderClient, oldAccount: Account, newAccount: Account)
```

Comment:
See above.

- Add a method to acquire the content provider client in the DataStore
- `updateAccount` will usually not be called during a sync, but as a standalone operation, so we can acquire the client directly in `updateAccount` and don't have to take it as argument.